desktop/libs/azure/src/azure/abfs/abfs.py

```diff
@@ -572,6 +572,15 @@ def rename(self, old, new):
     Renames a file
     """
     headers = {'x-ms-rename-source': '/' + urllib_quote(Init_ABFS.strip_scheme(old))}
+
+    # Required to sign the header with SAS token for RAZ
+    if RAZ.IS_ENABLED.get():
+      raz_http_client = RazHttpClient(self._user, self._url, exc_class=WebHdfsException, logger=LOG)
+      url = raz_http_client._make_url(headers['x-ms-rename-source'], params=None)
+
+      sas_token = raz_http_client.get_sas_token('PUT', self._user, url)
+      headers['x-ms-rename-source'] += '?' + sas_token
+
```

Comment:
Curious if all of this could be moved into AdlsRazClient#get_url()?
(i.e. to contain all the Raz specific there and not leak it here)

e.g. we look for `x-ms-rename-source` in the headers param there